### DISCUSSION TITLE
Define visit options on manual navigation

### DISCUSSION THREAD

**Discussion by bozdevs** [swup/swup#913] [General]
### Is there a way to access options on manual navigating?

![image](https://github.com/swup/swup/assets/57140595/88e32c59-e2f2-460f-8eec-19fe8d72d5a5)

In addition to the above options, Just like in this handler, I want to make some changes.
```js
swup.hooks.on('visit:start', (visit) => {
  visit.containers = ...;
  visit.animation.scope = ...;
  ...
});
```

For example; 

```js
swup.navigate(
  "/hello-world/", 
  { 
      containers: ["main"], // from ["main", "header"],
      animationScope: "containers", // from "html"
  }
);
```

**Comment by hirasso** [comment]
Intersting idea! Let us think about it.

**Comment by daun** [comment]
Can you elaborate on why using a hook doesn‘t work in your case? If you only need to edit the options once, you can use `swup.hooks.once()`. To conditionally edit the visit options, you can always wrap your modifications inside the hook in an `if` condition or two.

```js
swup.hooks.once('visit:start', (visit) => {
  visit.containers = ['main']
  visit.animation.scope = 'containers'
})

swup.navigate('/hello-world/')
```
